common/src/main/java/org/jboss/da/common/version/VersionAnalyzer.java

```diff
@@ -86,38 +82,40 @@ public VersionAnalysisResult analyseVersions(String querry, Collection<String> v
         return Optional.ofNullable(bestMatchVersion);
     }
 
-    private static boolean isDefaultSuffix(String suffix) {
-        return VersionParser.DEFAULT_SUFFIX.equals(suffix);
-    }
-
     /**
      * Assuming the two versions have the same OSGi representation, returns the more specific version. That means
      * X.Y.Z.something is preffered to X.Y.something which is preffered to X.something.
      */
     private String getMoreSpecificVersion(String first, String second) {
-        Pattern pattern = Pattern.compile("^" + VersionParser.RE_MMM + VersionParser.RE_QUALIFIER + "?");
-        Matcher firstMatcher = pattern.matcher(first);
-        Matcher secondMatcher = pattern.matcher(second);
+        Matcher firstMatcher = VERSION_PATTERN.matcher(first);
+        Matcher secondMatcher = VERSION_PATTERN.matcher(second);
         if (!firstMatcher.matches()) {
             throw new IllegalArgumentException("Couldn't parse version " + first);
         }
         if (!secondMatcher.matches()) {
             throw new IllegalArgumentException("Couldn't parse version " + second);
         }
-        if (firstMatcher.group("minor") == null && secondMatcher.group("minor") != null) {
-            return second;
+        boolean firstIsOSGi = first.equals(VersionParser.getOSGiVersion(first));
+        String firstMinor = firstMatcher.group("minor");
+        String firstMicro = firstMatcher.group("micro");
+        boolean returnFirst;
+
+        if (firstIsOSGi != second.equals(VersionParser.getOSGiVersion(second))) {
+            returnFirst = firstIsOSGi; // One of the version is not OSGi, prefer the OSGi version
+        } else if (!Objects.equals(firstMinor, secondMatcher.group("minor"))) {
```

Comment:
Not sure I understand this, couldn't it be that e.g. firstMinor is 3 and secondMinor is 5? The comment seems to contemplate only the presence of a null value as the reason why the !Objects.equals is true

Reply:
From the method documentation:
```java
    /**
     * Assuming the two versions have the same OSGi representation, returns the more specific version. That means
     * X.Y.Z.something is preffered to X.Y.something which is preffered to X.something.
     */
```
So it compares only versions that are "logically the same", so there can't be different number in firstMinor and secondMinor.
(This also means that if firstMinor is 3, then secondMinor is also always 3 and never null, because null can be "replaced" only by 0.)

Reply:
aaah got it now. thanks!